mssqlcli/mssql_cli.py

```diff
@@ -353,8 +340,36 @@ def execute_command(self, text, query):
             elif query.meta_changed:
                 self.refresh_completions(persist_priorities='all')
 
+        if not query.contains_secure_statement:
+            # Allow MssqlCompleter to learn user's preferred keywords, etc.
+            with self._completer_lock:
+                self.completer.extend_query_history(text)
+
+            self.query_history.append(query)
+
+    def execute_query(self, text, is_interactive=False):
```

Comment:
>is_interactive [](start = 34, length = 14)

I'm thinking we should make this an instance member that can be passed in via the options in the constructor. Then you don't have two duplicate methods. Defaulting to True of course.

Reply:
Let's sync on this—I explored this thoroughly and it's a bit complicated.